awx/ui_next/src/components/Lookup/Lookup.jsx

```diff
@@ -76,21 +80,33 @@ class Lookup extends React.Component {
     this.saveModal = this.saveModal.bind(this);
     this.getData = this.getData.bind(this);
     this.clearQSParams = this.clearQSParams.bind(this);
+    this.toggleDropdown = this.toggleDropdown.bind(this);
   }
 
   componentDidMount() {
-    this.getData();
+    Promise.all([this.getData()]);
   }
 
   componentDidUpdate(prevProps) {
-    const { location } = this.props;
-    if (location !== prevProps.location) {
+    const { location, selectedCategory } = this.props;
+    if (
+      location !== prevProps.location ||
+      prevProps.selectedCategory !== selectedCategory
+    ) {
       this.getData();
     }
   }
 
+  toggleDropdown() {
```

Comment:
I don't see this being used anywhere. I think you can delete it and the `isDropdownOpen` prop (AnsibleSelect manages its own dropdown open/closed state)